Yoast/wordpress-seo
# Update the upgrade routine to correctly clear the beta notification.

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the recalibration beta notification would still show after the beta ended. 

## Relevant technical choices:

* The upgrade routine did not correctly remove the notification from the database. This PR introduces a new function `clean_all_notifications` which clears all notification from the database field 'wp_yoast_notifications' in the table 'wp_usermeta'. 

## Test instructions

This PR can be tested by following these steps:

* Reproduce the issue as described in #12281. 
* Checkout this branch / install the build zip. The notification should be gone. 

* If this is difficult to test or you don't have an installation where the notification still shows you can bring it back to test it. Add the following function to `class-upgrade.php`
```
	/**
	 * Returns the notification.
	 *
	 * @return Yoast_Notification The notification for the notification center.
	 *
	 * @codeCoverageIgnore
	 */
	protected function get_notification() {
		$message = sprintf(
			/* translators: 1: link opening tag to the features page, 2: link closing tag, 3: Link to KB article, 4: expands to Yoast SEO */
			esc_html__(
				'We\'d love for you to try our new and improved %4$s analysis! Use the toggle on the %1$sFeatures tab%2$s in your %4$s settings. %3$sRead more about the new analysis%2$s.',
				'wordpress-seo'
			),
			'<a href="#top#features" onclick="jQuery(\'#features-tab\').click()">',
			'</a>',
			'<a href="' . WPSEO_Shortlinker::get( 'https://yoa.st/recalibration-beta-notice' ) . '" target="_blank">',
			'Yoast SEO'
		);

		$notification_options = array(
			'type'         => Yoast_Notification::WARNING,
			'id'           => 'wpseo-' . $this->notification_identifier,
			'priority'     => 1.0,
			'capabilities' => 'wpseo_manage_options',
		);

		return new Yoast_Notification( $message, $notification_options );
	}
```
* Add the following line to the `upgrade100()` method. 
```
Yoast_Notification_Center::get()->add_notification( $this->get_notification() );
```
* Use the Yoast test helper to switch versions and trigger the upgrade routine. 

## UI changes

## Quality assurance


Fixes #12281
